gst/nnstreamer/tensor_query/tensor_query_client.c

```diff
@@ -126,7 +257,391 @@ static void
 gst_tensor_query_client_get_property (GObject * object, guint prop_id,
     GValue * value, GParamSpec * pspec)
 {
-  return;
+  GstTensorQueryClient *self = GST_TENSOR_QUERY_CLIENT (object);
+
+  switch (prop_id) {
+    case PROP_SINK_HOST:
+      g_value_set_string (value, self->sink_host);
+      break;
+    case PROP_SINK_PORT:
+      g_value_set_int (value, self->sink_port);
+      break;
+    case PROP_SRC_HOST:
+      g_value_set_string (value, self->src_host);
+      break;
+    case PROP_SRC_PORT:
+      g_value_set_int (value, self->src_port);
+      break;
+    case PROP_PROTOCOL:
+      switch (self->protocol) {
+        case _TENSOR_QUERY_PROTOCOL_TCP:
+          g_value_set_string (value, "tcp");
+          break;
+        default:
+          break;
+      }
+      break;
+    case PROP_SILENT:
+      g_value_set_boolean (value, self->silent);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief Update src pad caps from tensors config.
+ */
+static void
+gst_tensor_query_client_update_caps (GstTensorQueryClient * self)
+{
+  GstTensorsConfig *config;
+  GstCaps *curr_caps, *out_caps;
+
+  config = &self->out_config;
+  out_caps = gst_tensor_pad_caps_from_config (self->srcpad, config);
+
+  /* Update src pad caps if it is different. */
+  curr_caps = gst_pad_get_current_caps (self->srcpad);
+  if (curr_caps == NULL || !gst_caps_is_equal (curr_caps, out_caps)) {
+    silent_debug_caps (self, out_caps, "set out-caps");
+    gst_pad_set_caps (self->srcpad, out_caps);
+  }
+
+  if (curr_caps)
+    gst_caps_unref (curr_caps);
+
+  gst_caps_unref (out_caps);
+}
+
+/**
+ * @brief This function handles sink event.
+ */
+static gboolean
+gst_tensor_query_client_sink_event (GstPad * pad,
+    GstObject * parent, GstEvent * event)
+{
+  GstTensorQueryClient *self = GST_TENSOR_QUERY_CLIENT (parent);
+
+  GST_DEBUG_OBJECT (self, "Received %s event: %" GST_PTR_FORMAT,
+      GST_EVENT_TYPE_NAME (event), event);
+
+  switch (GST_EVENT_TYPE (event)) {
+    case GST_EVENT_CAPS:
+    {
+      GstCaps *caps;
+      GstStructure *structure;
+
+      gst_event_parse_caps (event, &caps);
+      gst_pad_set_caps (self->srcpad, caps);
```

Comment:
Setting caps twice (L309) causes reconfigure for downstream elements. I'm not sure know how exactly this works but removing this line fixes my buggy pipeline test

Reply:
That's right. This line needs to be removed. Src pad caps should be set up after receiving the out config from the server.